scripts/validate-requirements.sh

```diff
@@ -0,0 +1,331 @@
+        ERRORS+=$'git default branch must be configured.\n'
+    fi
+}
+
+# Validate some utility tools that the environment must have before running the other checkers
+function validate_utils(){
+    if [ ! "$(command -v jq)" ]; then
+        echo_missing_installation "jq" "https://stedolan.github.io/jq/download/"
+        ERRORS+=$'jq not found.\n'
+    fi
+}
+
+# Validate the configuration of the Gcloud CLI
+function validate_gcloud_configuration(){
+
+    END_USER_CREDENTIAL_OUTPUT="$(gcloud config get-value account 2>&1 >/dev/null)"
+    if [ "$(echo "$END_USER_CREDENTIAL_OUTPUT" | grep -c unset)" -eq 1 ]; then
+        echo "You must configure an End User Credential."
+        echo "Visit https://cloud.google.com/sdk/gcloud/reference/auth/login and follow the instructions to authorize gcloud to access the Cloud Platform with Google user credentials."
+        ERRORS+=$'gcloud not configured with end user credential.\n'
+    fi
+
+    APPLICATION_DEFAULT_CREDENTIAL_OUTPUT="$(gcloud auth application-default print-access-token 2>&1 >/dev/null)"
+    if [ "$(echo "$APPLICATION_DEFAULT_CREDENTIAL_OUTPUT" | grep -c 'Could not automatically determine credentials')" -eq 1 ]; then
+        echo "You must configure an Application Default Credential."
+        echo "Visit https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login and follow the instructions to authorize gcloud to access the Cloud Platform with Google user credentials."
+        ERRORS+=$'gcloud not configured with application default credential.\n'
+    fi
+}
+
+#  Function to validate the roles attached to credentialled account
+function validate_credential_roles(){
+    check_org_level_roles "$END_USER_CREDENTIAL" "$ORGANIZATION_ID"
+    check_billing_account_roles "$END_USER_CREDENTIAL" "$BILLING_ACCOUNT"
+}
+
+# Verifies whether a user has the expected Organization level roles
+function check_org_level_roles(){
+
+    ORG_LEVEL_ROLES_OUTPUT=$(
+        gcloud organizations get-iam-policy "$2" \
+        --filter="bindings.members:$1" \
+        --flatten="bindings[].members" \
+        --format="table(bindings.role)" 2>/dev/null)
+
+    lines=0
+    for i in "${ORGANIZATION_LEVEL_ROLES[@]}"
+    do
+        if [[ "$ORG_LEVEL_ROLES_OUTPUT" == *"$i"* ]]; then
+            lines=$((lines + 1))
+        fi
+    done
+
+    if [ "$lines" -ne ${#ORGANIZATION_LEVEL_ROLES[@]} ]; then
+        echo "The User must have the Organization Roles resourcemanager.folderCreator, resourcemanager.organizationAdmin and roles/orgpolicy.policyAdmin"
+        ERRORS+=$'There are missing organization level roles on the Credential.\n'
+    fi
+}
+
+# Verifies whether a user has the expected Billing Level roles
+function check_billing_account_roles(){
+
+    BILLING_LEVEL_ROLES_OUTPUT=$(
+        gcloud beta billing accounts get-iam-policy "$2" \
```

Comment:
Looks like this may hang without warning if currently set project does not have `cloudbilling.googleapis.com` enabled. Can we require a project to be set and `cloudbilling.googleapis.com` enabled?

Reply:
So @bharathkkb I tested running this command with no default core/project set and it didn't raise any error/warning. 

Then I tested it with a default core/project but with the API cloudbilling.google.apis.com disabled and it didn't raise any error/warning as well. 

Is this the scenario you're refering to? If so, I couldn't reproduce the expected mentioned error.

Reply:
We can follow up if we get reports of this again. Thanks for investigating.